[BUG] @UtilityClass fails to make inner class static in Java 9+

**Describe the bug**

When a class is annotated with `@UtilityClass`, and this class has an inner class, the inner class is also treated as `static` in Java 1.8, but is not treated as `static` in Java 9 and later. This causes compilation failure under Java 9+ for code that compiled correctly in Java 1.8.

**To Reproduce**

Save the following as `Outer.java`:

```java
import lombok.experimental.UtilityClass;

@UtilityClass
class Outer {

  Object buildInner() {
    return new Inner();
  }

  class Inner {}
}
```

Compile this file using `javac` from Java 1.8, and observe that compilation succeeds with no diagnostic output:

```text
% /path/to/java-1.8/bin/javac -cp lombok.jar Outer.java
```

Now compile this file using `javac` from Java 9 or later, and observe that compilation fails:

```text
% /path/to/java-9/bin/javac -cp lombok.jar Outer.java
Outer.java:7: error: non-static variable this cannot be referenced from a static context
    return new Inner();
           ^
1 error
```

**Expected behavior**

Java 9 and later compilers should have compiled this source file, producing no diagnostic output.

**Version info (please complete the following information):**
 - Lombok version: 1.18.22
 - Platform: `javac` 9

**Additional context**

The problem first arises with Java 9, but also occurs in Java 11, 14, and 17. (I haven't tested Java 10, 12, 13, 15, or 16.)

#2519 seems closely related, but was closed because it could not be reproduced. However, that earlier report mentioned nothing about different Java versions. In this new report, I find that the Java compiler version is critical: the problem only arises in Java 9 and later, not in Java 1.8.

--- a/src/core/lombok/javac/handlers/HandleUtilityClass.java
+++ b/src/core/lombok/javac/handlers/HandleUtilityClass.java
@@ -25,6 +25,9 @@ import static lombok.core.handlers.HandlerUtil.handleExperimentalFlagUsage;
 import static lombok.javac.handlers.JavacHandlerUtil.*;
 
 import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.code.Type.ClassType;
 import com.sun.tools.javac.tree.JCTree.JCAnnotation;
 import com.sun.tools.javac.tree.JCTree.JCBlock;
 import com.sun.tools.javac.tree.JCTree.JCClassDecl;
@@ -124,6 +127,10 @@ public class HandleUtilityClass extends JavacAnnotationHandler<UtilityClass> {
 			} else if (element.getKind() == Kind.TYPE) {
 				JCClassDecl innerClassDecl = (JCClassDecl) element.get();
 				innerClassDecl.mods.flags |= Flags.STATIC;
+				ClassSymbol innerClassSymbol = innerClassDecl.sym;
+				if (innerClassSymbol != null && innerClassSymbol.type instanceof ClassType) {
+					((ClassType) innerClassSymbol.type).setEnclosingType(Type.noType);
+				}
 			}
 		}
 		

--- a/test/transform/resource/after-delombok/UtilityClass.java
+++ b/test/transform/resource/after-delombok/UtilityClass.java
@@ -2,8 +2,13 @@ final class UtilityClass {
 	private static long someField = System.currentTimeMillis();
 	static void someMethod() {
 		System.out.println();
+		new InnerClass();
+		new InnerStaticClass();
 	}
 	protected static class InnerClass {
+		private String innerInnerMember;
+	}
+	protected static class InnerStaticClass {
 		private String innerInnerMember;
 	}
 	@java.lang.SuppressWarnings("all")

--- a/test/transform/resource/after-ecj/UtilityClass.java
+++ b/test/transform/resource/after-ecj/UtilityClass.java
@@ -5,11 +5,19 @@ final @lombok.experimental.UtilityClass class UtilityClass {
       super();
     }
   }
+  protected static class InnerStaticClass {
+    private String innerInnerMember;
+    protected InnerStaticClass() {
+      super();
+    }
+  }
   private static long someField = System.currentTimeMillis();
   <clinit>() {
   }
   static void someMethod() {
     System.out.println();
+    new InnerClass();
+    new InnerStaticClass();
   }
   private @java.lang.SuppressWarnings("all") UtilityClass() {
     super();

--- a/test/transform/resource/before/UtilityClass.java
+++ b/test/transform/resource/before/UtilityClass.java
@@ -4,9 +4,15 @@ class UtilityClass {
 	
 	void someMethod() {
 		System.out.println();
+		new InnerClass();
+		new InnerStaticClass();
 	}
 	
 	protected class InnerClass {
+		private String innerInnerMember;
+	}
+	
+	protected static class InnerStaticClass {
 		private String innerInnerMember;
 	}
 }

